### PR TITLE
Reorganized Contracted Hat Saves.

### PR DESCRIPTION
**Pull request by dataswifty**
I refactored the contract data save.

### CODE REVIEW

File: hat/app/org/hatdex/hat/api/controllers/RichData.scala [hunk 1]
```diff
@@ -1011,7 +1025,7 @@ class RichData @Inject() (
 
         requestIsAllowed.flatMap { testResult =>
           testResult match {
-            case Right(RequestVerified(ns)) =>
+            case Right(RequestVerified(ns @ _)) =>
```
[the remark this answers is not in the document]
Comment (dataswifty) [reply hat/app/org/hatdex/hat/api/controllers/RichData.scala:1028]: You can name the variable that is extracted, but if you don't need it, adding the suffix `@ _` means the compiler won't complain.

File: hat/app/org/hatdex/hat/api/controllers/RichData.scala [hunk 2]
```diff
@@ -717,89 +717,105 @@ class RichData @Inject() (
     UserAwareAction.async(parsers.json[ContractRequestBody]) {
       implicit request =>
```
[the remark this answers is not in the document]
Comment (dataswifty) [reply hat/app/org/hatdex/hat/api/controllers/RichData.scala:718]: Yup, agreed.

File: hat/app/org/hatdex/hat/api/controllers/RichData.scala [hunk 3]
```diff
@@ -717,89 +717,105 @@ class RichData @Inject() (
...
-              Future.successful(NotFound)
+            Future.successful(NotFound)
+          }
+
+          // Body exists, so verify the request and process the body
+          case Some(jsonBody) => {
+            usersService.getUser(contractRequestBody.hatName).flatMap {
+              hatUser =>
+                hatUser match {
+                  case None => {
```
[the remark this answers is not in the document]
Comment (dataswifty) [reply hat/app/org/hatdex/hat/api/controllers/RichData.scala:734]: You should have seen it before. :). 

Will do.

File: hat/app/org/hatdex/hat/api/controllers/RichData.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (benshaw) [reply hat/app/org/hatdex/hat/api/controllers/RichData.scala:1028]: How is this different than just _

**Comment by benshaw** [comment]
@dataswifty where is this at ?

**Comment by dataswifty** [comment]
> @dataswifty where is this at ?

Not done yet.

**Comment by benshaw** [comment]
@ty *bump* i want to ensure this doesn't get forgotten

**Pull request merged**